Fix for https://github.com/wso2/streaming-integrator/issues/165

dilini-muthumala opened the pull request:
## Purpose
Improves the fix done for https://github.com/wso2/streaming-integrator/issues/165 by returning false when FileSystemException occurs.
Also 
- resolves https://github.com/wso2/streaming-integrator/issues/168
- resolves https://github.com/siddhi-io/siddhi-io-file/issues/121
- resolves https://github.com/siddhi-io/siddhi-io-file/issues/124
- resolves https://github.com/wso2/streaming-integrator/issues/183

dilini-muthumala committed:
Fix for https://github.com/wso2/streaming-integrator/issues/165

dilini-muthumala committed:
Fix for https://github.com/wso2/streaming-integrator/issues/168

dilini-muthumala committed:
Fix for https://github.com/wso2/streaming-integrator/issues/168

ramindu90 commented on component/src/main/java/io/siddhi/extension/io/file/FileSource.java:
```diff
@@ -618,11 +617,23 @@ public void pause() {
+            this.scheduledFuture = siddhiAppContext.getScheduledExecutorService().
+                    scheduleAtFixedRate(fileSourcePoller, 0, 1, TimeUnit.SECONDS);
+        }
+        if (isTailingEnabled && fileSourceConfiguration.getFileServerConnector() != null) {
```
If tailing disabled, and if its a large file which will take time more than the persisting interval, the file processing may stop. Isnt it?
Shall we use [filePointer](https://github.com/siddhi-io/siddhi-io-file/blob/4858827171969d749751e3f580fa6dee853c83a8/component/src/main/java/io/siddhi/extension/io/file/FileSource.java#L951) and resolve that issue?
This may affect for the files that will be processing in a given dir.uri as well

dilini-muthumala replied:
Hi Ramindu, 
In the existing pause() method, the file processing thread is not stopped which is the reason why I thought that there is nothing to resume in the resume() method, assuming that the file will be processed anyway. But let's check this further, to make sure that happens. I have created a separate issue to track this particular issue: https://github.com/siddhi-io/siddhi-io-file/issues/120.

Thanks

dilini-muthumala committed:
Return false when the file does not exist

dilini-muthumala committed:
Merge branch 'master' of github.com:siddhi-io/siddhi-io-file

ramindu90 committed:
Resolves https://github.com/siddhi-io/siddhi-io-file/issues/124

Resolves following issues in file:move():
'include.by.regexp' and 'exclude.root.dir' parameters in file:move() are not dynamic
When 'exclude.root.dir' parameter is set to true, the destination path does not get created correctly
When four parameters are given, the regex is always set to ''

ramindu90 committed:
Fix documentation issue which refer to the path as URI

ramindu90 committed:
Fix for https://github.com/wso2/streaming-integrator/issues/183
